feat: Upri resources trimming (#16036)
unoplatform/uno: doc/articles/features/upri-trimming.md

---

> uid: Uno.Features.UpriTrimming

Reviewer: ```suggestion
uid: Uno.Features.StringResourcesTrimming
```

---

> # UPRI resource trimming

Reviewer: ```suggestion
# String Resource Trimming
```

> # UPRI resource trimming
> 
> UPRI resource trimming is an optional feature used to reduce the final payload size of an Uno Platform application.
> 
> It trims unused localization resources from Uno.
> 
> This feature is currently available for WebAssembly projects.

Reviewer: ```suggestion
# String Resource Trimming

String resource trimming (also known as upri trimming) is an optional feature used to reduce the final payload size of an Uno Platform application by removing unused localization languages throughout the assemblies referenced by the app.

This feature is currently available for the `net8.0-webassembly` target framework and `.Wasm` legacy projects.
```

## Using UPRI resource trimming

To enable resources trimming, you need to declare which culture(s) your application uses. In your csproj add:

```xml
<ItemGroup>

>     <UnoSupportedLanguage Include="en-US" />

Reviewer: Let's add a few more lines for developers that may not understand itemgroups
